infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/impl/cachehandler/HotRodCacheCreationHandler.java

```diff
@@ -0,0 +1,106 @@
+
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.exceptions.HotRodClientException;
+import org.infinispan.commons.configuration.XMLStringConfiguration;
+
+/**
+ * Creates {@link org.infinispan.client.hotrod.RemoteCache} if necessary,
+ * apply a configuration to the cache, if required,
+ * verifies the existence of any configuration
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class HotRodCacheCreationHandler implements HotRodCacheHandler {
+
+	public static final String OGM_BASIC_CONFIG =
+		"<infinispan><cache-container>" +
+		"	<distributed-cache-configuration name=\"configuration\">" +
+		"     <locking striping=\"false\" acquire-timeout=\"10000\" concurrency-level=\"50\" isolation=\"READ_COMMITTED\"/>" +
+		"     <transaction mode=\"NON_DURABLE_XA\" />" +
+		"     <expiration max-idle=\"-1\" />" +
+		"     <indexing index=\"NONE\" />" +
+		"     <state-transfer timeout=\"480000\" await-initial-transfer=\"true\" />" +
+		"   </distributed-cache-configuration>" +
+		"</cache-container></infinispan>";
+
+	private static final Log log = LoggerFactory.make( MethodHandles.lookup() );
+
+	private final Map<String, String> cacheConfigurationByName;
+
+	private Set<String> failedCacheConfigurationNames = new HashSet<>();
+
+	public HotRodCacheCreationHandler(String globalConfiguration, Map<String, String> perCacheConfiguration) {
+		this.cacheConfigurationByName = perCacheConfiguration;
+
+		// applying globalConfiguration to all caches where a configuration is not defined
+		perCacheConfiguration.forEach( ( cache, configuration ) -> {
+			if ( configuration == null ) {
+				perCacheConfiguration.put( cache, globalConfiguration );
+			}
+		} );
+	}
+
+	@Override
+	public void startAndValidateCaches(RemoteCacheManager hotrodClient) {
+
+		failedCacheConfigurationNames.clear();
+
+		cacheConfigurationByName.entrySet().forEach( entry -> {
+			if ( entry.getValue() == null ) {
+				startAndValidateCache( hotrodClient, entry.getKey() );
+			}
+			else {
+				startAndValidateCache( hotrodClient, entry.getKey(), entry.getValue() );
+			}
+		} );
+
+		if ( failedCacheConfigurationNames.size() > 1 ) {
+			throw log.expectedCacheConfiguratiosNotDefined( failedCacheConfigurationNames );
+		}
+		if ( failedCacheConfigurationNames.size() == 1 ) {
+			throw log.expectedCacheConfigurationNotDefined( failedCacheConfigurationNames.iterator().next() );
+		}
+	}
+
+	protected void startAndValidateCache(RemoteCacheManager hotrodClient, String cacheName, String cacheConfiguration) {
+
+		try {
+			hotrodClient.administration()
+				.getOrCreateCache( cacheName, cacheConfiguration );
+		}
+		catch ( HotRodClientException ex ) {
+			failedCacheConfigurationNames.add( cacheConfiguration );
+		}
+	}
+
+	protected void startAndValidateCache(RemoteCacheManager hotrodClient, String cacheName) {
+
+		hotrodClient.administration()
+			.getOrCreateCache( cacheName, new XMLStringConfiguration( OGM_BASIC_CONFIG ) );
+	}
```

Comment:
It is diffent: in case of using a configuration defined on the server, this configuration should be present, in case of applying our Hibernate OGM tempalte we dont need any configuration server side. From this difference born the idea to catch the runtime expection only in case of using a defined configuration.

Reply:
Looking at this again it worries me a bit that we don't log the exception that caused the problem. I understand that most of the time is going to be a non defined configuration but I assume there could be other situations that could cause problems. Is it possible to distinguish between the two cases?

Reply:
I'll try